src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -927,14 +849,17 @@ export function useSwapInputsController({
             })
           );
 
+          const prevOutputNativeValue = inputValues.value.outputNativeValue;
+          const prevInputAmount = inputValues.value.inputAmount;
+          const newOutputAmount = outputNativePrice > 0 ? divWorklet(prevOutputNativeValue, outputNativePrice) : prevInputAmount;
+
           inputValues.modify(values => {
             return {
               ...values,
               inputAmount,
-              inputNativeValue: inputValues.value.inputNativeValue,
-              outputAmount:
-                outputNativePrice > 0 ? divWorklet(inputValues.value.outputNativeValue, outputNativePrice) : inputValues.value.inputAmount,
-              outputNativeValue: inputValues.value.outputNativeValue,
+              inputNativeValue: mulWorklet(newInputAmount, inputNativePrice),
```

Comment:
Updated `input/outputNativeValue' to use the `mulWorklet` so that it is more precise and based on the new inputAmount and the new native price.